src/modules/push-notifications/services/push.notifications.service.ts

```diff
@@ -0,0 +1,130 @@
+import { Injectable } from '@nestjs/common';
+import { pushNotificationsConfig, scAddress } from 'src/config';
+import { MXApiService } from 'src/services/multiversx-communication/mx.api.service';
+import { ApiConfigService } from 'src/helpers/api.config.service';
+import {
+    AddressUtils,
+    BinaryUtils,
+    ErrorLoggerAsync,
+} from '@multiversx/sdk-nestjs-common';
+
+import {
+    ContractKeysRaw,
+    NotificationConfig,
+    NotificationResult,
+    NotificationType,
+} from '../models/push.notifications.types';
+import { PushNotificationsSetterService } from './push.notifications.setter.service';
+import { XPortalApiService } from 'src/services/multiversx-communication/mx.xportal.api.service';
+
+@Injectable()
+export class PushNotificationsService {
+    constructor(
+        private readonly apiService: MXApiService,
+        private readonly apiConfigService: ApiConfigService,
+        private readonly notificationsSetter: PushNotificationsSetterService,
+        private readonly xPortalApiService: XPortalApiService,
+    ) {}
+
+    @ErrorLoggerAsync()
+    async usersWithEnergyFromContractStorage(): Promise<string[]> {
+        const contractAddress = scAddress.simpleLockEnergy;
+        const contractKeysRaw: ContractKeysRaw =
+            await this.apiService.doGetGeneric(
+                'getContractKeys',
+                `address/${contractAddress}/keys`,
+            );
+
+        const contractPairs = Object.entries(
+            contractKeysRaw?.data?.pairs || {},
+        );
+
+        const userEnergyKey = BinaryUtils.stringToHex('userEnergy');
+        const userEnergyKeys = contractPairs
+            .filter(([key, _]) => key.startsWith(userEnergyKey))
+            .map(([key, _]) => key.replace(userEnergyKey, ''));
+
+        const userEnergyAddresses = userEnergyKeys.map((key) =>
+            AddressUtils.bech32Encode(key),
+        );
+
+        return userEnergyAddresses;
+    }
+
+    async sendNotifications(
+        addresses: string[],
+        notificationType: NotificationType,
+    ): Promise<NotificationResult> {
+        return this.sendNotificationsInBatches(
```

Comment:
What is the difference between this method and `sendNotificationsInBatches`. This one seems just a wrapper for the other.